scripts/create_registry_schema.py

```diff
@@ -0,0 +1,365 @@
+        if production != "production":
+            old_col = columns["input_production_id"]
+            fkey = ForeignKey(f"{production}.dataset.dataset_id")
+            new_input_production_id = Column(old_col.name,
+                                             old_col.type,
+                                             fkey)
+            del columns["input_production_id"]
+            columns["input_production_id"] = new_input_production_id
+
+    # Table metadata
+    meta = {"__tablename__": "dependency",
+            "__table_args__": {"schema": schema}}
+
+    Model = type(class_name, (Base,), {**columns, **meta})
+    return Model
+
+
+# The following should be adjusted whenever there is a change to the structure
+# of the database tables.
+_DB_VERSION_MAJOR = 2
+_DB_VERSION_MINOR = 2
+_DB_VERSION_PATCH = 0
+_DB_VERSION_COMMENT = "Add `location_type` for dataset table"
+
+# Parse command line arguments
+parser = argparse.ArgumentParser(
+    description="""
+Creates dataregistry tables for specified schema and connection information (config)""",
+    formatter_class=argparse.ArgumentDefaultsHelpFormatter,
+)
+parser.add_argument(
+    "--schema",
+    help="name of schema to contain tables. Will be created if it doesn't already exist",
+    default=f"{SCHEMA_VERSION}",
+)
+parser.add_argument(
+    "--production-schema", default="production",
+    help="name of schema containing production tables.",
+)
+parser.add_argument("--config", help="Path to the data registry config file")
+
+args = parser.parse_args()
+schema = args.schema
+prod_schema = args.production_schema
+
+# Connect to database to find out what the backend is
+db_connection = DbConnection(args.config, schema)
+if db_connection.dialect == "sqlite":
+    if schema == "production":
+        raise ValueError("Production not available for sqlite databases")
+    # In fact we don't use schemas at all for sqlite
+    schema = None
+else:
+    if schema != prod_schema:
+        # production schema, tables must already exists and schema
+        # major and minor versions must match
+        stmt = f"select db_version_major, db_version_minor from {prod_schema}.provenance order by provenance_id desc limit 1"
+        try:
+            with db_connection.engine.connect() as conn:
+                result = conn.execute(text(stmt))
+                result = pd.DataFrame(result)
+                conn.commit()
+        except Exception:
+            raise RuntimeError("production schema does not exist or is ill-formed")
+        if result["db_version_major"][0] != _DB_VERSION_MAJOR | int(result["db_version_minor"][0]) > _DB_VERSION_MINOR:
+            raise RuntimeError("production schema version incompatible")
+
+if schema:
+    stmt = f"CREATE SCHEMA IF NOT EXISTS {schema}"
+    with db_connection.engine.connect() as conn:
+        conn.execute(text(stmt))
+        conn.commit()
+
+    # Grant reg_reader access
+    try:
+        with db_connection.engine.connect() as conn:
```

Comment:
`reg_writer` is only for production? (should it also be granting write privileges here?)

Could be in a loop over `reg_reader` and `reg_writer` to save space

Reply:
If it's not production then the account creating the schema would normally already be `reg_writer`, so it shouldn't have to give itself privilege (though I suppose it doesn't hurt). If it isn't normal - that is, it's being created from an account other than `reg_writer`, it's some kind of private testing thing which probably no one other than the creator is going to use.